xbmc/settings/lib/SettingUpdate.cpp

```diff
@@ -14,8 +14,12 @@
 #include "utils/XBMCTinyXML.h"
 #include "utils/log.h"
 
-CSettingUpdate::CSettingUpdate() : CStaticLoggerBase("CSettingUpdate")
+Logger CSettingUpdate::s_logger;
+
+CSettingUpdate::CSettingUpdate()
 {
+  if (s_logger == nullptr)
```

Comment:
Can this race? are `CSettingUpdate` objects ever constructed in parallel on first use?

Reply:
For the settings classes this is hard to tell because they are also used for add-on settings which are constructed whenever an add-on needs them. So it can probably also happen in parallel.

But IMO it doesn't really matter whether this part of the code races or not. `CLog::GetLogger()` or rather `spdlog::get()` is thread-safe and the second call to it will just return the same logger as the first call. So even if the next line is executed twice it's not a big deal. It may be a bit suboptimal from a performance point of view but adding a lock here which is processed on every call to `GetLogger()` would be worse.